### PR TITLE
Utilize `HAS_VTARG_READ` defined in avrdude.conf

### PR DESCRIPTION
**Pull request by MCUdude**
Resolves issue #1469.

Tested with the following programmers, where I've tried with and without `HAS_VTARG_READ` in avrdude.conf.

* `xplainedpro`
* `xplainedpro_pdi`
* `xplainedpro_updi`
* `powerdebugger_jtag`
* `powerdebugger_isp`
* `powerdebugger_pdi`
* `powerdebugger_updi`
* `dragon_isp`
* `dragon_pp`
* `avrispmkii`
* `xplainedmini_updi`
* `xplainedmini_isp`

### CODE REVIEW

File: src/stk500v2.c [hunk 1]
```diff
@@ -3718,20 +3724,22 @@ static void stk500v2_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp
...
-    fmsg_out(fp, "%sSCK period      : %.2f us\n", p,
-	    (float) (sck_stk600 + 1) / 8.0);
-    stk500v2_getparm2(pgm, PARAM2_CLOCK_CONF, &clock_conf);
-    oct = (clock_conf & 0xf000) >> 12u;
-    dac = (clock_conf & 0x0ffc) >> 2u;
-    f = pow(2, (double)oct) * 2078.0 / (2 - (double)dac / 1024.0);
-    f = f_to_kHz_MHz(f, &unit);
-    fmsg_out(fp, "%sOscillator      : %.3f %s\n",
-            p, f, unit);
+    fmsg_out(fp, "%sSCK period      : %.2f us\n", p, (float) (sck_stk600 + 1) / 8.0);
```
Comment (stefanrueger) [inline src/stk500v2.c:3734]: Small comment. The pre-existing cast `(float)` is superfluous: Owing to the division by 8.0 the nominator is cast to `double` anyway, and `%f` always expects a double. Actually, all `(float)` casts in stk500v2.c and jtag3.c are superfluous by the same reasoning.

**Comment by stefanrueger** [comment]
@MCUdude Code looks great. You have tested it. So, I am happy for this PR to go ahead.

**Comment by mcuee** [comment]
This PR looks good to me.

1) STK500 clone with official V1 FW, no regression from git main.
```
mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude_pr1470 -C ./avrdude_pr1470.conf -c stk500v1 -P /dev/ttyUSB0 -p m16 -t
avrdude_pr1470: AVR device initialized and ready to accept instructions
avrdude_pr1470: device signature = 0x1e9403 (probably m16)

avrdude_pr1470: processing -t interactive terminal
avrdude> parms
Vtarget         : 5.3 V
Varef           : 5.2 V
Oscillator      : 3.686 MHz
SCK period      : 1.1 us
avrdude> quit

avrdude_pr1470 done.  Thank you.

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude -c stk500v1 -P /dev/ttyUSB0 -p m16 -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9403 (probably m16)

avrdude: processing -t interactive terminal
avrdude> parms
Vtarget         : 5.3 V
Varef           : 5.2 V
Oscillator      : 3.686 MHz
SCK period      : 1.1 us
avrdude> quit

avrdude done.  Thank you.
```

2) STK500v 2 clone with unofficial FW (fake support of the extra features).
```
mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude_pr1470 -C ./avrdude_pr1470.conf -c stk500 -P /dev/ttyUSB1 -p m16 -t
avrdude_pr1470: AVR device initialized and ready to accept instructions
avrdude_pr1470: device signature = 0x1e9403 (probably m16)

avrdude_pr1470: processing -t interactive terminal
avrdude> parms
Vtarget         : 4.7 V
SCK period      : 2.2 us
Varef           : 2.5 V
Oscillator      : 230.400 kHz
avrdude> quit

avrdude_pr1470 done.  Thank you.

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude -c stk500 -P /dev/ttyUSB1 -p m16 -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9403 (probably m16)

avrdude: processing -t interactive terminal
avrdude> parms
Vtarget         : 4.7 V
SCK period      : 2.2 us
Varef           : 2.5 V
Oscillator      : 230.400 kHz
avrdude> quit

avrdude done.  Thank you.
```

**Comment by mcuee** [comment]
No regressions for PICKit 4 in ISP mode.
```
mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude -c pickit4_isp -p m8a -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9307 (probably m8a)

avrdude: processing -t interactive terminal
avrdude> parms
SCK period      : 8.00 us
Vtarget         : 4.90 V

avrdude> quit

avrdude done.  Thank you.

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude_pr1470 -C ./avrdude_pr1470.conf -c pickit4_isp -p m8a -t
avrdude_pr1470: AVR device initialized and ready to accept instructions
avrdude_pr1470: device signature = 0x1e9307 (probably m8a)

avrdude_pr1470: processing -t interactive terminal
avrdude> parms
SCK period      : 8.00 us
Vtarget         : 4.90 V

avrdude> quit

avrdude_pr1470 done.  Thank you.
```

**Comment by mcuee** [comment]
No regressions with JTAG ICE 1 either. Tested with two clones of different designs

1) Better JTAG ICE 1 clone
```

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude_pr1470 -C ./avrdude_pr1470.conf -P /dev/ttyUSB0 -c jtag1 -p m128a -t
avrdude_pr1470: AVR device initialized and ready to accept instructions
avrdude_pr1470: device signature = 0x1e9702 (probably m128a)

avrdude_pr1470: processing -t interactive terminal
avrdude> parms
Vtarget       : 4.5 V
JTAG clock    : 1 MHz (1.0 us)
avrdude> quit

avrdude_pr1470 done.  Thank you.

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude -P /dev/ttyUSB0 -c jtag1 -p m128a -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9702 (probably m128a)

avrdude: processing -t interactive terminal
avrdude> parms
Vtarget       : 4.5 V
JTAG clock    : 1 MHz (1.0 us)
avrdude> quit

avrdude done.  Thank you.
```

2) Another clone which reads the voltage wrongly
```
mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude_pr1470 -C ./avrdude_pr1470.conf -P /dev/ttyUSB0 -c jtag1 -p m128a -t
avrdude_pr1470: AVR device initialized and ready to accept instructions
avrdude_pr1470: device signature = 0x1e9702 (probably m128a)

avrdude_pr1470: processing -t interactive terminal
avrdude> parms
Vtarget       : 6.2 V
JTAG clock    : 250 kHz (4.0 us)
avrdude> quit

avrdude_pr1470 done.  Thank you.

mcuee@UbuntuSwift3:~/build/avr/avrdude_bin$ ./avrdude -P /dev/ttyUSB0 -c jtag1 -p m128a -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9702 (probably m128a)

avrdude: processing -t interactive terminal
avrdude> parms
Vtarget       : 6.2 V
JTAG clock    : 250 kHz (4.0 us)
avrdude> quit

avrdude done.  Thank you.
```

**Comment by mcuee** [comment]
@MCUdude and @stefanrueger 

I think this PR is good to go.

**Pull request merged**